### PR TITLE
#160451521 user should be able to track comments history

### PR DESCRIPTION
**Pull request by kampkelly**
#### What does this PR do?
This pull request adds keeping records of comments history

#### Description of Task to be completed?
- Write tests for tracking comment history
- Write the code to track comment history

#### How should this be manually tested?
Using postman and as a signed in user, make requests to the following api endpoints.

To update a comment: /api/v1/articles/:articleSlug/comments/commentId/edit -> PUT request (params: comment)
To retrieve comment history for a user: /api/v1/articles/:articleSlug/comments/:commentId/myhistory -> GET request
To retrieve comment history for all users: /api/v1/articles/:articleSlug/comments/:commentId/allhistory -> GET request

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#160451521
#### Questions:```
None

### CODE REVIEW

File: server/controllers/commentHistoryController.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  */
+  static getMyCommentHistory(req, res) {
+    ArticleComment.findOne({
+      where: { articleSlug: req.params.articleSlug, id: req.params.commentId },
+    }).then((comment) => {
+      if (comment) {
+        const searchParams = {
+          commentId: req.params.commentId, articleSlug: req.params.articleSlug, userId: req.user.id
+        };
+        getCommentHistory(searchParams, res, 'comment history retrieved', 'you have no histories for this comment');
```
Comment (kampkelly) [inline server/controllers/commentHistoryController.js:41]: This is calling a helper function. The same block of code was used in multiple functions in this controller. So I moved it to a helper function. The strings sent along are the specific response messages for this function.

File: server/controllers/commentHistoryController.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  */
+  static getAllCommentHistory(req, res) {
+    ArticleComment.findOne({
+      where: { articleSlug: req.params.articleSlug, id: req.params.commentId },
+    }).then((comment) => {
+      if (comment) {
+        const searchParams = {
+          commentId: req.params.commentId, articleSlug: req.params.articleSlug
+        };
+        getCommentHistory(searchParams, res, 'all comment histories retrieved', 'there are no histories for this comment');
```
Comment (kampkelly) [inline server/controllers/commentHistoryController.js:63]: This is calling a helper function. The same block of code was used in multiple functions in this controller. So I moved it to a helper function. The strings sent along are the specific response messages for this function.

**Review by slimsolz — COMMENTED** [review]
LGTM

File: server/helpers/article.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 
 export const computeOffset = (req) => {
   const page = Object.keys(req.query).length ? parseInt(req.query.page, 10) : 1;
+  // eslint-disable-next-line
```
Comment (kampkelly) [inline server/helpers/article.js:4]: Eslint is throwing an error on isNaN ---> 'no-restricted-globals'. So I disabled the check for this line.

**Review by samdiano — COMMENTED** [review]
LGTM

File: server/tests/commentsHistory.models.test.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import { expect } from 'chai';
+import db from '../models';
+
+
+const validCommentHistory = {
```
Comment (s-akinrele) [inline server/tests/commentsHistory.models.test.js]: move it to another file

File: server/tests/commentsHistory.test.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+/* eslint-disable prefer-destructuring */
+import faker from 'faker';
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../index';
+
+chai.use(chaiHttp);
+
+const user = {
+  username: faker.internet.userName(),
```
Comment (s-akinrele) [inline server/tests/commentsHistory.test.js]: and this

**Review by s-akinrele — CHANGES_REQUESTED** [review]
I had some comments

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**